### PR TITLE
[gatsby-plugin-vercel-builder] Move "public" dir instead of copying

### PR DESCRIPTION
**Pull request by TooTallNate**
We are getting reports of some larger Gatsby projects running out of disk space during build which seem to be the result of copying the static directory multiple times.

So move instead of copying, which doesn't fully solve the problem, but it does kick the can down the road a bit. Very large projects would probably still hit the disk limit at some point. Anyways, moving is faster than copying so this is still a net positive.

### CODE REVIEW

File: packages/gatsby-plugin-vercel-builder/src/handlers/build.ts [hunk 1]
```diff
@@ -88,10 +87,6 @@ export async function copyFunctionLibs({
         src: join('.cache', 'page-ssr'),
         dest: join(functionDir, '.cache', 'page-ssr'),
       },
-      // {
-      //   src: join(functionDir, '.cache', 'query-engine', 'assets'),
-      //   dest: join(functionDir, 'assets'),
-      // },
```
Comment (styfle) [inline packages/gatsby-plugin-vercel-builder/src/handlers/build.ts:94]: Should this comment be removed?

Comment (TooTallNate) [reply packages/gatsby-plugin-vercel-builder/src/handlers/build.ts:94]: Ya, was just cleaning up some dead code while in here.

Comment (cb1kenobi) [reply packages/gatsby-plugin-vercel-builder/src/handlers/build.ts:94]: According to https://github.com/vercel/vercel/commit/ae89b8b8bee068cf9499867d6c794cb2ccc47c24#diff-3b89128206f28715de44e9a753ba57ac89f9ea0b484035ca0d74c0a3f8d9204bR98-R101, these lines have always been commented out. Sounds like they're not needed.

**Review by cb1kenobi — DISMISSED** [review]

File: packages/gatsby-plugin-vercel-builder/src/helpers/static.ts [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
     'static',
     prefix ?? ''
   );
-  await ensureDir(targetDir);
-  await copy(join(process.cwd(), 'public'), targetDir);
+  await ensureDir(dirname(targetDir));
+  try {
+    await move(publicDir, targetDir);
+  } catch (err: any) {
+    console.log('failed to move "public" dir', err);
```
Comment (styfle) [inline packages/gatsby-plugin-vercel-builder/src/helpers/static.ts]:
```suggestion
    console.error('failed to move "public" dir', err);
```

**Review by styfle — DISMISSED** [review]

File: packages/gatsby-plugin-vercel-builder/src/helpers/static.ts [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
     'static',
     prefix ?? ''
   );
-  await ensureDir(targetDir);
-  await copy(join(process.cwd(), 'public'), targetDir);
+  await ensureDir(dirname(targetDir));
+  try {
+    await move(publicDir, targetDir);
+  } catch (err: any) {
+    console.error('failed to move "public" dir', err);
```
Comment (EndangeredMassa) [inline packages/gatsby-plugin-vercel-builder/src/helpers/static.ts]: **suggestion (non-blocking):**

```suggestion
    console.error(`Failed to move "public" dir from "${publicDir}" to "${targetDir}". Copying instead.`, err);
```

**Review by EndangeredMassa — DISMISSED** [review]

**Review by Ethan-Arrowood — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**